### ISSUE TITLE
Extension not showing till I refresh the playlist.

### ISSUE THREAD

**Issue by iaendi** [nrednav/youtube-playlist-duration-calculator#52]
<img width="996" alt="image" src="https://github.com/nrednav/youtube-playlist-duration-calculator/assets/130339496/0d3d9501-7393-4148-bd25-ff8e095fdc54">

Everytime I enter the “Watch Later” playlist, I can't see the extension UI at first glance.

<img width="912" alt="image" src="https://github.com/nrednav/youtube-playlist-duration-calculator/assets/130339496/6375707f-cf19-4c75-a2db-3e7d1f129744">
I need to refresh the page, then the UI shows up.

**Comment by MisterTickle** [comment]
I have the same issue here.
Chrome Version 124.0.6367.202 (Official Build) (64-bit)

**Comment by nrednav** [comment]
Hi @iaendi and @MisterTickle, that's odd, I'm not able to reproduce the same issue.

Tested on Firefox 125.0.1 (64-bit) and Chrome 124.0.6367.208 (Official Build) (64-bit)

Few questions about the issue:
- When does it happen? Is it when you visit a playlist via the address bar or via the sidebar?
- How often does it happen?
- Does it only happen for the Watch Later playlist or any kind of playlist?

When the UI doesn't load, can you please check the browser console for any messages related to the extension.

To access the browser console:
- Chrome
  - Mac = Cmd + Option + J
  - Windows = Ctrl + Shift + J
- Firefox
  - Mac = Cmd + Shift + J
  - Windows = Ctrl + Shift + J
  
You should see something like this:
- Chrome
  ![image](https://github.com/nrednav/youtube-playlist-duration-calculator/assets/26251262/852723cd-6e9b-43ce-902a-a5cffdb9f004)
- Firefox 
  ![image](https://github.com/nrednav/youtube-playlist-duration-calculator/assets/26251262/504af48c-bb6c-453b-bcc5-c7f1d3910e95)

**Comment by MisterTickle** [comment]
Hi,
I've been hoping it would happen so that I could reply with something tangible. For me, it's pretty rare, so I'm not very worried about it.

Pretty often, my Watch Later page loads but with the videos all missing from the right hand column. Re-loading fixes it. I think this is YT's fault and not this extension, and it's a much more pressing issue :)

For me, it's rare for the WL page to load **without** the Duration Calculator's information in the middle column. It did happen a few minutes ago which prompted this message because I wanted to check the Console.

There was nothing I could see in the Console on the time the page loaded without the extension loading. 
The only errors there at that time were all similar to the ones I get when I load it and the extension works, which are things like this:

```
866[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive. See <URL>
desktop_polymer.js:8029 [Violation] 'yt-page-data-fetched' handler took 1265ms
26[Violation] 'requestAnimationFrame' handler took <N>ms
/feed/history:1 requestStorageAccessFor: Permission denied.
/feed/history:1 requestStorageAccessFor: Permission denied.
/feed/history:1 requestStorageAccessFor: Permission denied.
desktop_polymer.js:8029 [Violation] 'yt-page-data-fetched' handler took 489ms
[Violation] Forced reflow while executing JavaScript took 46ms
playlist:1 The resource https://i.ytimg.com/generate_204 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

```

**Comment by iaendi** [comment]
@nrednav Sorry for the slow response. 
My environment is: MacBook M1 Pro 16 inch + Chrome (Version 124.0.6367.208 (Official Build) (arm64)).
In my case, I always clicked on "Watch Later"(playlist) from the YouTube homepage sidebar, then the extension wouldn't show.
I checked browser console:
<img width="863" alt="image" src="https://github.com/nrednav/youtube-playlist-duration-calculator/assets/130339496/b8ed3db5-0528-4f89-9659-05fb71741217">

Then I refreshed immediately:
<img width="843" alt="image" src="https://github.com/nrednav/youtube-playlist-duration-calculator/assets/130339496/223ac84a-ac7d-4ef5-bf90-2c73527275db">

This issue occurs every time when I do "Youtube Homepage--Sidebar--Watch Later"
And automatically solved when I refresh the page at least for one time.
After that, I close the tab, repeat the step, extension works perfectly, even after I clear the cache and try a second time.



Meanwhile, when the extension didn't show, I tried to directly open another playlist in the address bar, everything looks normal.
<img width="763" alt="image" src="https://github.com/nrednav/youtube-playlist-duration-calculator/assets/130339496/719b7ee2-9b72-4c86-bd7e-7bb85d50ab18">


To sum it up, something stopped the extension from loading the playlist when I visit the “Watch Later” playlist from the sidebar, but only for the first visit in a period of time (maybe daily, didn't count, but very frequently)

**Comment by nrednav** [comment]
Hi @iaendi @MisterTickle, I've just released v2.1.3 which should hopefully address this issue. After it's approved by the Chrome & Firefox stores, please do give it a try when you can and let me know if there is any improvement.

**Comment by MisterTickle** [comment]
Thanks very much!

**Comment by iaendi** [comment]
Thanks for the news update.
I installed v2.1.3 from the Chrome web store and it's working perfectly.
Really appreciate the work you did.
Thank you
